### PR TITLE
bpo-42246: Remove DO_NOT_EMIT_BYTECODE macros, so that while loops and if statements conform to PEP 626.

### PR DESCRIPTION
**Pull request by markshannon**
PEP 626 requires all expressions and statements to be traced, even if the code is redundant.
This PR does that for `while` and `if` statements.
The performance impact on `while 1:` loops is minimized by copying the test to the end of the loop.

```
while test:
     body
```
used to compile to:
```
start:
    if not test: goto end
    body
    goto start
end:
```
It now compiles to:
```
    if not test: goto end
loop:
    body
    if test goto loop
end:
```
which will be a tiny bit slower for `while 1` loops and a bit faster for other `while` loops.

I'm skipping news, as this PR is part of PEP 626 and is covered by that.

https://bugs.python.org/issue42246

### CODE REVIEW

File: Lib/test/test_peepholer.py [hunk 1]
```diff
@@ -409,21 +409,6 @@ def f(cond1, cond2):
         self.assertLessEqual(len(returns), 6)
         self.check_lnotab(f)
 
-    def test_elim_jump_after_return2(self):
```
Comment (markshannon) [inline Lib/test/test_peepholer.py:412]: I've removed this test, as doesn't test what it claims to test. There is no dead code in
```
def f(cond1, cond2):
    while 1:
        if cond1: return 4
```

**Comment by markshannon** [comment]
The three failures are, in order:
* An asyncio test timeout. Comments about the reliability of asyncio are for another issue.
* A pre-existing failure.
* Missing symbols in LTO, possibly caused by a (wrong) warning about buffer overflow when resizing a bytes object.

No leaks, or real failures, so I'm merging.

**Review by Sassybabydoll1969 — COMMENTED** [review]
<SPAM>

**Review by bazilio43 — COMMENTED** [review]
<SPAM>

**Review by bazilio43 — COMMENTED** [review]
[Андрей ](https://www.facebook.com/profile.php?id=100035662340112)

**Pull request merged**